### DISCUSSION TITLE
New mastodon user experience is too difficult, lets solve the emptiness together.

### DISCUSSION THREAD

**Discussion by privacytoolsIO** [mastodon/mastodon#27397] [General]
### Motivation
We've started our Mastodon instance a week ago ([social.privacytools.io](https://social.privacytools.io/)) and it's going well so far: 400+ users and 1000+ statuses. But we've noticed the same pattern with most users: They are excited and sign up, they see no content and 80% give up after a couple of minutes. 20% of users will stick around and push through and look actively for content and users to follow. But it's tough and takes forever. Let's brainstorm and find solutions to give new users a better experience.

### Pitch
Today we came up with a quick solution:
https://social.privacytools.io/about/more#follow

It's a simple HTML list of who to follow, using the rules feature in the site settings and a couple of lines of custom CSS. It would be nice if we could do that via admin panel.

I'm aware there is also a service called [Trunk](https://communitywiki.org/trunk/) to mass follow people. But thats hard to find as a new user. Even our follow list on the about page is not the right place.

A longterm solution would be a better connectivity and an immediate "Who to follow?" feature after signing up.

I'm aware of #642 and  #4870 but these issues are years old now, and I think this became more urgent since "bridge.joinmastodon.org" is down now.

Edit: Here is an active discussion going on already
https://octodon.social/@kensanata/101900811439802416

**Comment by jelv** [comment]
There is a feature called federation relay. You can fill your federated timeline with content from other instances. So you could fill it up with toots from selected instances you think are interesting for your audience. https://github.com/tootsuite/mastodon/pull/7998

**Comment by ghost** [comment]
The PTIO Twitter page is not pointing people to mastdon, and seems quite active, giving little reason for people to leave twitter.  A bit of push and pull effect would help.  Make the twitter account reference the mastodon account, and make the Mastodon one more active.

BTW, some accounts could be followed from a GS instance but not the social.privacytools.io Mastodon instance:

https://status.fsf.org/fsf
https://gs.jonkman.ca/bobjonkman/
https://freespeechextremist.com/users/se7en/

**Comment by jonaharagon** [comment]
> There is a feature called federation relay. You can fill your federated timeline with content from other instances. 

You cannot fill your timeline with *any* other instance, only from other instances who have joined your/another relay server. Which seems unfortunate, you should be able to subscribe to instances directly.

**Comment by marchgithub** [comment]
You should look to Pleroma's implementation next time. All Pleroma instances can be followed as a relay by default, and it's easy enough to turn it off for your instance if that's not your thing. Would be great if Mastodon did that too.

**Comment by privacytoolsIO** [comment]
@libBletchley Thanks, good idea. I've changed the Twitter profile already to point people to Mastodon. https://twitter.com/privacytoolsIO

Here is feedback from a frustrated new user:

> well.. I expected more than 3 people on here, and @cloudfare spamming 20 posts in a row. Sigh... back to reddit. There is at least a pretense of up to 100 people whom I can talk about privacy and stuff... Never used or got the appeal of Twitter, so maybe I had wrong expectations... Cool concept, nevertheless.. Just... lifeless... Same with Linux distros. Fragmentation just spreads out focus and communities, and everyone does their own thing. Might be good for free speech, though. Or not.

Source: https://social.privacytools.io/web/statuses/101918520858880009

**Comment by ghost** [comment]
There is never a GNU Social or Mastodon timeline that offers significantly resourceful content without mirroring Twitter.  No highly-followed people are willing to sacrifice Twitter.  This makes Twitter a **superset** of GS and Mastodon and that's a problem.  Imagine if Snowden were to say "I'm off Twitter now.  Follow me on Mastodon".  People would.

## campaign freedom-focused and net neutrality-respecting orgs to downplay Twitter

Freedom-respecting orgs that should be advocating GS and Mastodon are falling short of their duty.  EFF has always had "unofficial" on their Mastodon profile, implying that you'd be better informed to follow them on Twitter.  Perhaps a campaign to get mirrored microblogs to:

* declare that their federated account is "official"
* declare that their Twitter account is "unofficial"
* link to their federated timeline in Twitter, but don't link back to Twitter

## campaign to get public schools and offices to used federated microblogs

Twitter requires Tor users to supply a mobile phone number to register.  This makes the Twitter walled-garden exclusive.  It's utterly despicable that taxpayer-funded public schools like LAUSD use Twitter, thus forcing students, parents, and teachers to comply with the privacy-hostile demands of a private corporation in order to interact with their school.  It's a First Amendment abuse in the US, IMO.

We need a movement to get schools and other tax-funded agencies to stop promoting and using Twitter exclusively.  Twitter should be secondary to a federated open public microblog.

## add privacy features Twitter doesn't have

When someone quips an URL that links to a CloudFlare site, it's a bad practice and it's ultimately detrimental to privacy and net neutrality.  The social.privacytools.io instance might consider solving the problem by detecting such messages and interrupt the user with a warning about why it's a bad idea.  The user could be given the option to have the URL replaced with an archived or cached link.  If they decide to proceed with the CloudFlare link anyway, social.privacytools.io could have a robotic post warning readers that it's a dodgy link.

A privacy-focused feature such as that would add value to social.privacytools.io.  It would have to be done in a way that doesn't break the API.. so a web user would get the warning but probably not someone using bitlbee.

**Comment by abeorch** [comment]
As a newbie and I'm no doubt about to tread on some toes as I'm probably not privy to previous conversations. I'm going to suggest that the emptiness problem could be resolved by allowing cross instance exchange of search queries with a like for like exchange of query resource with appropriate throttling/limiting to avoid smaller instances being overwhelmed.  Similarly scope of toots could allow posters to decide whether they want content to be instance only, global or indeed allow them to decide from which instances users could find their content. 

Relay servers at first look seem to me like a kind of administrative curation of selected content that puts alot of work on the admin to decide what their users might want to see. 

Allowing exchange of cross instance searches either based on #hashtag or full text means that users could reach out and find content across the federation. This might be contentious since I'm not sure whether Mastodon seeks to be a federation or confederation in its organisational model.

**Comment by todoneunl** [comment]
There's a 3rd party opt-in user directory here: https://communitywiki.org/trunk/, although it needs some UI love. And the federation relay should be the way to go to fill your federated timeline. Unfortunately it's difficult to find relay servers and the software isn't really maintained right now (see https://source.joinmastodon.org/mastodon/pub-relay).
